Postfix increment/decrement on immutable parameters not caught at compile-time or runtime

## Summary

Postfix increment (`++`) and decrement (`--`) operators on immutable function parameters are not caught at compile-time or runtime. The operation silently does nothing instead of producing an error.

## Reproduction

```ez
import @std
using std

do test_int(x int) {
    println("Before: ${x}")
    x++  // Should fail - x is immutable
    println("After: ${x}")
}

do main() {
    test_int(5)
}
```

## Expected Behavior

The typechecker should catch this at compile-time with error E5016:
```
error[E5016]: cannot modify immutable variable 'x' (declared as const or as non-& parameter)
```

## Actual Behavior

- Code compiles successfully
- Code runs without error
- The `x++` operation silently does nothing (output shows "Before: 5" and "After: 5")

## Notes

- Regular assignment (`x = x + 1`) and compound assignment (`x += 1`) ARE correctly caught at compile-time
- Only postfix operators bypass the check

## Affected Code

Likely in `pkg/typechecker/typechecker.go` - need to add mutability check for `PostfixExpression` statements.

--- a/integration-tests/fail/errors/E5016_postfix_immutable_param.ez
+++ b/integration-tests/fail/errors/E5016_postfix_immutable_param.ez
@@ -1,0 +1,17 @@
+/*
+ * Error Test: E5016 - postfix on immutable parameter
+ * Expected: "cannot modify"
+ * Regression test for bug #539
+ */
+
+do test_increment(x int) {
+    x++  // ERROR: cannot modify immutable param with postfix
+}
+
+do test_decrement(y int) {
+    y--  // ERROR: cannot modify immutable param with postfix
+}
+
+do main() {
+    test_increment(5)
+}

--- a/pkg/errors/codes.go
+++ b/pkg/errors/codes.go
@@ -168,6 +168,7 @@ var (
 	E5020 = ErrorCode{"E5020", "range-in-operand-not-integer", "value checked against range must be integer"}
 	E5021 = ErrorCode{"E5021", "panic", "explicit panic called"}
 	E5022 = ErrorCode{"E5022", "assertion-failed", "assertion condition was false"}
+	E5023 = ErrorCode{"E5023", "postfix-requires-integer", "postfix operator needs integer operand"}
 )
 
 // =============================================================================

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -2189,7 +2189,8 @@ func elementsEqual(a, b Object) bool {
 func evalPostfixExpression(node *ast.PostfixExpression, env *Environment) Object {
 	ident, ok := node.Left.(*ast.Label)
 	if !ok {
-		return newError("postfix operator requires identifier")
+		return newErrorWithLocation("E5015", node.Token.Line, node.Token.Column,
+			"postfix operator %s requires a variable identifier", node.Operator)
 	}
 
 	val, ok := env.Get(ident.Value)
@@ -2199,7 +2200,8 @@ func evalPostfixExpression(node *ast.PostfixExpression, env *Environment) Object
 
 	intVal, ok := val.(*Integer)
 	if !ok {
-		return newError("postfix operator requires integer")
+		return newErrorWithLocation("E5023", node.Token.Line, node.Token.Column,
+			"postfix operator %s requires integer operand, got %s", node.Operator, val.Type())
 	}
 
 	var newVal *big.Int

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -1295,6 +1295,10 @@ func (tc *TypeChecker) checkExpression(expr ast.Expression) {
 
 	case *ast.RangeExpression:
 		tc.checkRangeExpression(e)
+
+	case *ast.PostfixExpression:
+		tc.checkExpression(e.Left)
+		tc.checkPostfixExpression(e)
 	}
 }
 
@@ -1327,6 +1331,24 @@ func (tc *TypeChecker) checkRangeExpression(rangeExpr *ast.RangeExpression) {
 	tc.checkExpression(rangeExpr.End)
 	if rangeExpr.Step != nil {
 		tc.checkExpression(rangeExpr.Step)
+	}
+}
+
+// checkPostfixExpression validates postfix operators (++ and --)
+// These operators modify the operand, so we must check mutability
+func (tc *TypeChecker) checkPostfixExpression(postfix *ast.PostfixExpression) {
+	// Check mutability - error if trying to modify an immutable variable
+	if rootVar := tc.extractRootVariable(postfix.Left); rootVar != "" {
+		isMutable, found := tc.isVariableMutable(rootVar)
+		if found && !isMutable {
+			line, column := tc.getExpressionPosition(postfix.Left)
+			tc.addError(
+				errors.E5016,
+				fmt.Sprintf("cannot modify immutable variable '%s' (declared as const or as non-& parameter)", rootVar),
+				line,
+				column,
+			)
+		}
 	}
 }
 

